server/handler/base.go

```diff
@@ -193,5 +196,41 @@ func (b *Base) EvaluateFetchedConfig(ctx context.Context, prctx pull.Context, cl
 		return errors.Errorf("evaluation resulted in unexpected state: %s", result.Status)
 	}
 
+	if performActions && statusState == "pending" && !prctx.IsDraft() {
+		// Intentionally kept to a small result size, since we just want to determine if there are existing reviewers
+		subsetCurrentReviewers, _, err := client.PullRequests.ListReviewers(ctx, prctx.RepositoryOwner(), prctx.RepositoryName(), prctx.Number(), &github.ListOptions{
+			Page:    0,
+			PerPage: 10,
+		})
+		if err != nil {
+			logger.Warn().Err(err).Msg("Unable to list request reviewers")
+		}
+
+		if subsetCurrentReviewers != nil && len(subsetCurrentReviewers.Users) == 0 && len(subsetCurrentReviewers.Teams) == 0 {
+			r := rand.New(rand.NewSource(time.Now().UnixNano()))
+			requestedUsers, err := reviewer.FindRandomRequesters(ctx, prctx, result, r)
+			if err != nil {
+				logger.Warn().Err(err).Msg("Unable to select random request reviewers")
+			}
+
+			if len(requestedUsers) > 0 {
+				reviewers := github.ReviewersRequest{
+					Reviewers:     requestedUsers,
+					TeamReviewers: []string{},
+				}
+
+				logger.Debug().Msgf("PR is not in draft, there are no current reviewers, and reviews are requested from %q users", requestedUsers)
+				_, _, err = client.PullRequests.RequestReviewers(ctx, prctx.RepositoryOwner(), prctx.RepositoryName(), prctx.Number(), reviewers)
+				if err != nil {
+					logger.Warn().Err(err).Msg("Unable to request reviewers")
+				}
+			} else {
+				logger.Debug().Msg("No users found for review")
+			}
+		} else {
+			logger.Debug().Msg("PR is not in draft or there are existing reviewers, not adding anyone.")
+		}
+	}
+
 	return b.PostStatus(ctx, prctx, client, statusState, statusDescription)
```

Comment:
I'm wondering if the status should be posted first, before doing any reviewer evaluations. Depending on how many list calls we have to make, selecting reviewers could delay user feedback for a couple seconds, particularly if GitHub is slow to respond to requests.